### PR TITLE
Try to solve boss_jeklik

### PR DESCRIPTION
**Pull request by w5860363**
Try using a different https://github.com/TrinityCore/TrinityCore/pull/27274 Solutions https://github.com/TrinityCore/TrinityCore/issues/27259

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 1]
```diff
@@ -195,6 +228,8 @@ struct boss_jeklik : public BossAI
 
         DoMeleeAttackIfReady();
     }
+private:
+    uint32 uiSummon_counter;
```
Comment (MaxtorCoder) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:232]:
```suggestion
    uint32 _summonCounter;
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 2]
```diff
@@ -180,9 +208,14 @@ struct boss_jeklik : public BossAI
...
-                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 0.f, true))
-                        if (TempSummon* flyingBat = me->SummonCreature(NPC_FRENZIED_BAT, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ() + 15.0f, 0.0f, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15s))
-                            flyingBat->AI()->AttackStart(target);
+                    if (uiSummon_counter < 3)
+                    {
+                        Talk(SAY_RAIN_FIRE);
+                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 0.f, true))
+                            if (TempSummon* flyingBat = me->SummonCreature(NPC_FRENZIED_BAT, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ() + 15.0f, 0.0f, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15s))
+                                flyingBat->AI()->AttackStart(target);
+                        ++uiSummon_counter;
```
Comment (MaxtorCoder) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:217]:
```suggestion
                        ++_summonCounter;
```

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp [hunk 3]
```diff
@@ -180,9 +208,14 @@ struct boss_jeklik : public BossAI
                     events.ScheduleEvent(EVENT_GREATER_HEAL, 25s, 35s, 0, PHASE_TWO);
                     break;
                 case EVENT_SPAWN_FLYING_BATS:
-                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 0.f, true))
-                        if (TempSummon* flyingBat = me->SummonCreature(NPC_FRENZIED_BAT, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ() + 15.0f, 0.0f, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15s))
-                            flyingBat->AI()->AttackStart(target);
+                    if (uiSummon_counter < 3)
```
Comment (MaxtorCoder) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp:211]:
```suggestion
                    if (_summonCounter < 3)
```

**Review by MaxtorCoder — CHANGES_REQUESTED** [review]

**Comment by offl** [comment]
This PR does opposite to what is done in #27274
And you need to test such changes locally before opening a PR or at least test them locally after opening a PR

**Pull request closed without merging**